src/hex.rs

```diff
@@ -0,0 +1,37 @@
+use crate::{Bytes, BytesMut};
+use std::fmt::{Formatter, LowerHex, Result, UpperHex};
+
+struct BytesRef<'a>(&'a [u8]);
+
+impl<'a> LowerHex for BytesRef<'a> {
```

Comment:
Since it anyway needs macros to impl for multiple types at once, wouldn't be easier to just define impl without `BytesRef`?

Reply:
In this case, we can merge `BsDebug` and `BytesRef` later, and avoid code bloat.

Reply:
Let me know if my understanding is wrong.

Reply:
I suspect in the end it would be inlined, but now I think about it, it is a little difference